app/src/main/java/org/dhis2/usescases/searchTrackEntity/SearchTEPresenter.java

```diff
@@ -413,7 +414,9 @@ public void downloadTei(String teiUid, String enrollmentUid) {
                                         view.showBreakTheGlass(teiUid, enrollmentUid);
                                         break;
                                     default:
-                                        view.displayMessage(new D2ErrorUtils(view.getContext()).getErrorMessage(t));
+                                        view.displayMessage(
```

Comment:
Why don't you inject D2ErrorUtils in the presenter?

Reply:
Done! I've injected the resource manager as D2ErrorUtils requires the wrapperContext to provide the correct translation.